src/alertmanager.py

```diff
@@ -106,6 +105,8 @@ def __init__(
         self._web_config_path = web_config_path
         self._is_tls_enabled = tls_enabled
 
+        self._peer_addresses = _parse_peer_addresses(peer_addresses, self._ha_port)
```

Comment:
`_get_peer_addresses` in `charm.py` is only used for this.
It renders addresses with the ha port, even though the ha port is passed as an arg to `WorkloadManager`.

Also, `web_external_url` is a deprecated config option so we do not expect to have a path component in the peer address.

So, I think we should refactor `_get_peer_addresses` to return the hostname. That could obviate the helpers added in this PR.

It would be great if we could use a consistent terminology for url parts, e.g. like [here](https://docs.python.org/3/library/urllib.parse.html).

Reply:
Looking at this a bit more, I'm not sure refactoring removes helpers so much as moves them around.

* `private_address` in the peer relation is `SCHEME://FQDN:API_PORT`
* scrape config tagets needs `FQDN:API_PORT` (no scheme)
* `WorkloadManager` needs `FQDN:HA_PORT` (no scheme)

So if we don't change `private_address`'s content, we still need helpers to remove scheme and change port.  

I've done another pass emphasizing reducing the code changes.  wdyt?  I think we still need some of these address conversions either way.